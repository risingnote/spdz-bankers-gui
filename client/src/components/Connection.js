/**
 * Responsible for displaying SPDZ connection information based on props passed in.
 */
import React from 'react';
import { List } from 'immutable'
import ProxyStatus from './ProxyStatus'

const Connection = (props) => {
  const combinedComponentStyle = Object.assign(
    {
      padding: '10px',
      paddingTop: '0',
      border: '1px solid darkgrey',
      margin: '0 auto',
      textAlign: 'left'
    }, props.componentStyle)

  const combinedHeadingStyle = Object.assign(
    {
      color: 'rgb(28, 118, 152)',
      marginRight: '10px'
    }, props.headerStyle)

  const proxyServers = (proxyStatusList) => {
    return proxyStatusList.map( (proxyServer) => {
      return <ProxyStatus key={proxyServer.get('url')} status={proxyServer.get('status')} 
                url={proxyServer.get('url')} urlStyle={props.urlStyle} />
    })
  }

  return (
    <div style={combinedComponentStyle}>
      <p style={combinedHeadingStyle}>Spdz Proxy Servers</p>
      {proxyServers(props.proxyStatusForDisplay)}
    </div>
  )
} 

Connection.propTypes = {
  proxyStatusForDisplay: React.PropTypes.instanceOf(List).isRequired,
  componentStyle: React.PropTypes.object,
  headerStyle: React.PropTypes.object,
  urlStyle: React.PropTypes.object
}

export default Connection