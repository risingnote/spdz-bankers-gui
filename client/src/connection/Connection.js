/**
 * Responsible for displaying SPDZ connection information based on props passed in.
 */
import React from 'react';
import { List } from 'immutable'
import ProxyStatus from './ProxyStatus'
import './Connection.css'

const Connection = (props) => {
  const proxyServers = (proxyServerList) => {
    return proxyServerList.map( (proxyServer) => {
      return <ProxyStatus key={proxyServer.get('url')} status={proxyServer.get('status')} url={proxyServer.get('url')} />
    })
  }

  return (
    <div className='Connection-panel'>
      <p className='Connection-subHead'>Spdz Proxy Servers</p>
      {proxyServers(props.spdzProxyServerList)}
    </div>
  )
} 

Connection.propTypes = {
  spdzProxyServerList: React.PropTypes.instanceOf(List).isRequired
}

export default Connection