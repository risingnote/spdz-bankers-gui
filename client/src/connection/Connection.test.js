import React from 'react'
import renderer from 'react-test-renderer'

import Connection from './Connection'
import { twoProxiesWith1Connected } from '../test_support/ProxyServerList'

describe('Setup display component rendering', () => {
  it('Renders as expected (compared to a snapshot) when passed properties', () => {

    const tree = renderer.create(
      <Connection spdzProxyServerList={twoProxiesWith1Connected} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
