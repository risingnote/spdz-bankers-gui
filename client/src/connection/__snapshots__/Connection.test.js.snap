exports[`Setup display component rendering Renders as expected (compared to a snapshot) when passed properties 1`] = `
<div
  className="Connection-panel">
  <p
    className="Connection-subHead">
    Spdz Proxy Servers
  </p>
  <div
    className="ProxyStatus-row">
    <div
      className="ProxyStatus-col1 ProxyStatus-marker-disconnected" />
    <div
      className="ProxyStatus-col2 smallText">
      http://spdzProxy.one:4000
    </div>
  </div>
  <div
    className="ProxyStatus-row">
    <div
      className="ProxyStatus-col1 ProxyStatus-marker-connected" />
    <div
      className="ProxyStatus-col2 smallText">
      http://spdzProxy.two:4000
    </div>
  </div>
</div>
`;
